### ISSUE TITLE
`printer` has invalid spec

### ISSUE THREAD

**Issue by bhb** [bhb/expound#191]
Repro:

```clojure
(deftest a-test
  (st/instrument)
  (binding [s/*explain-out* expound/printer]
    (s/explain-str :foobar/name "")
    ))
```

`lein test`

`s/*explain-out*` should control the success message, not the outer functions

https://github.com/clojure/spec.alpha/blob/5a4834eeb2967b2eca382efe0afcb3f590bdb3c2/src/main/clojure/clojure/spec/alpha.clj#L257

Workaround is to redefine the spec:

```clojure
(s/fdef expound/printer
        :args (s/cat :explain-data (s/nilable map?))
        :ret nil?)
```

**Comment by bhb** [comment]
@RickMoynihan @aviflax This has been fixed in Expound 0.8.5.

**Comment by aviflax** [comment]
Fantastic, thank you!
